doc/book/features/middleware-types.md

```diff
@@ -82,7 +139,7 @@ One note: the `$request` argument does not require a typehint, and examples
 throughout the manual will omit the typehint when demonstrating callable
 middleware.
 
-## Double-pass middleware
+## Double-pass middleware (Expressive 1.X and 2.X)
```

Comment:
Should add a note that it is supported in 3.x via `DoublePassMiddlewareWrapper`?